packages/core/lib/generators/integrations/image_integration.dart

```diff
@@ -161,11 +179,47 @@ ${isPackage ? "\n  static const String package = '$packageName';" : ''}
         FileInput(File(asset.fullPath)),
       );
       final size = result.size;
-      return ImageMetadata(size.width.toDouble(), size.height.toDouble());
+      final animation = parseAnimation ? _parseAnimation(asset) : null;
+
+      return ImageMetadata(
+        width: size.width.toDouble(),
+        height: size.height.toDouble(),
+        animation: animation,
+      );
     } catch (e) {
       stderr
           .writeln('[WARNING] Failed to parse \'${asset.path}\' metadata: $e');
     }
     return null;
   }
+
+  ImageAnimation? _parseAnimation(AssetType asset) {
+    final decoder = switch (asset.mime) {
+      'image/gif' => img.GifDecoder(),
+      'image/webp' => img.WebPDecoder(),
+      _ => null,
+    };
+
+    if (decoder == null) {
+      return null;
+    }
+
+    final file = File(asset.fullPath);
+    final bytes = file.readAsBytesSync();
+    final image = decoder.decode(bytes);
```

Comment:
Will it be better if we continue by returning a null result and printing logs when an image fails to be decoded?

Reply:
Added a try...catch block to handle exceptions.